cmake/clang-tidy.cmake

```diff
@@ -0,0 +1,117 @@
+if(CMAKE_GENERATOR MATCHES "Ninja|Unix Makefiles")
+    message(STATUS "🔧 Compile command will be used to enable clang-tidy")
+    set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+else()
+    message(WARNING "🚧 Compile command can't be used because the CMAKE_GENERATOR is ${CMAKE_GENERATOR}.
+You have to use Ninja or Unix Makefiles.
+Without compile command, clang-tidy will not work.")
+endif()
```

Comment:
Can those messages be explicit and explain what the "Compile command" here is?

Reply:
Ok 👍